test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import logging
+import time
+import pytest
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.random_tables import RandomTables
+from test.pylib.rest_client import inject_error
+from test.pylib.util import unique_name, wait_for_cql_and_get_hosts
+from test.topology.conftest import skip_mode
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_table_desc_read_barrier(request, manager: ManagerClient) -> None:
+    """
+    Regression test for #19213.
+
+    Test that the schema is always updated after read barrier.
+
+    Arrange:
+        - start 2 nodes
+        - create a table
+        - inject error (sleep) on the first node
+        - disable the schema agreement wait
+    Act:
+        - alter the table (change schema)
+        - trigger the read barrier
+    Assert:
+        - the schema retrieved on both hosts matches
+    """
+    servers = [await manager.server_add() for _ in range(2)]
+
+    logging.info("Waiting until driver connects to every server")
+    cql = manager.get_cql()
+    h1, h2, *_ = await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+
+    logging.info("Creating a table")
+    random_tables = RandomTables(request.node.name, manager, unique_name(), 1)
+    table = await random_tables.add_table(ncolumns=5, pks=1)
```

Comment:
I'd recommend just writing down the CQL statement with some predefined simple schema instead of using the `RandomTables` utility which in this case only obfuscates the test a bit IMO.